src/core/friendly_errors/sketch_reader.js

```diff
@@ -0,0 +1,272 @@
+    for (let i = 0; i < variables.length; i++) {
+      if (constants[variables[i]] !== undefined) {
+        let url = `https://p5js.org/reference/#/p5/${variables[i]}`;
+        p5._friendlyError(
+          translator('fes.sketchReaderErrors.reservedConst', {
+            url: url,
+            symbol: variables[i]
+          })
+        );
+        break;
+      }
+    }
+  };
+
+  /**
+   * Takes a list of functions defined by the user in the code
+   * as an array and checks if the list contains a reserved p5.js function.
+   * If found then display a friendly error message.
+   *
+   * @method checkForFuncs
+   * @private
+   * @param {Array} funcs list of functions defined by user
+   */
+  const checkForFuncs = funcs => {
+    const p5Constructors = {};
+    for (let key of Object.keys(p5)) {
+      // Get a list of all constructors in p5. They are functions whose names
+      // start with a capital letter
+      if (typeof p5[key] === 'function' && key[0] !== key[0].toLowerCase()) {
+        p5Constructors[key] = p5[key];
+      }
+    }
+    for (let i = 0; i < funcs.length; i++) {
+      //for every function name obtained check if it matches any p5.js function name
+      const keyArr = Object.keys(p5Constructors);
+      let j = 0;
+      for (; j < keyArr.length; j++) {
+        if (p5Constructors[keyArr[j]].prototype[funcs[i]] !== undefined) {
+          //if a p5.js function is used ie it is in the funcs array
+          p5._friendlyError(
+            translator('fes.sketchReaderErrors.reservedFunc', {
+              symbol: funcs[i]
+            })
+          );
+          break;
+        }
+      }
+      if (j < keyArr.length) break;
+    }
+  };
+
+  /**
+   * Takes an array in which each element is a line of code
+   * containing a variable definition(Eg: arr=['let x = 100', 'const y = 200'])
+   * and extracts the variables defined.
+   *
+   * @method extractVariables
+   * @private
+   * @param {Array} arr array of lines of code
+   */
+  const extractVariables = arr => {
+    //extract variable names from the user's code
+    let matches = [];
+    arr.forEach(ele => {
+      //extract a, b, c from let/const a=10, b=20, c;
+      if (ele.includes(',')) {
+        matches.push(
+          ...ele.split(',').flatMap(s => {
+            let match;
+            if (s.includes('=')) {
+              match = s.match(/(\w+)\s*(?==)/i);
+              return match[1];
+            } else if (!s.match(new RegExp('[[]{}]'))) {
+              let m = s.match(/(?:(?:let|const|var)\s+)?([\w$]+)/);
+              if (m !== null)
+                return s.match(/(?:(?:let|const|var)\s+)?([\w$,]+)/)[1];
+            } else return [];
+          })
+        );
+      } else {
+        //extract a from let/const a=10;
+        const reg = /(?:(?:let|const)\s+)([\w$]+)/g;
+        const found = ele.matchAll(reg);
+        for (const match of found) {
+          matches.push(match[1]);
+        }
+      }
+    });
```

Comment:
I suggest splitting this code a bit: You can try creating a function from the lambda at line 104, and end up with something like:

```javascript
const extractVariables = arr => arr
        .map(applyVariableRegex) // Apply regexes to check for matches (lines 104 to 113, kind of)
        .filter(...) // Filter out whatever doesn't match (line 113)
        .reduce(...) // Flatten the "ele" items into the final list

const applyVariableRegex = s => {
  if (s.includes('=')) { ... }
  ...
}

...
```

Reply:
the `extractVariables` function's sole purpose is to apply the regex and extract the function names so I guess code should be here and using another function for regex matching would be overdoing it. I am happy to defer though. What are your thoughts?

Reply:
I agree with @Aloneduckling on this one!

Reply:
Alright! That makes sense